kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/ErrorCollector.kt

```diff
@@ -2,24 +2,59 @@
 
 package io.kotest.assertions
 
-import kotlinx.coroutines.asContextElement
+import kotlinx.coroutines.CopyableThreadContextElement
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlin.coroutines.CoroutineContext
 
-actual val errorCollector: ErrorCollector get() = ThreadLocalErrorCollector.instance.get()
+actual val errorCollector: ErrorCollector get() = threadLocalErrorCollector.get()
 
 /**
- * A [kotlin.coroutines.CoroutineContext.Element] which keeps the error collector synchronized with thread-switching coroutines.
+ * A [CoroutineContext.Element] which keeps the error collector synchronized with thread-switching coroutines.
  *
  * When using [withClue] or [assertSoftly] on the JVM without the Kotest framework, this context element
  * should be added to each top-level coroutine context, e.g. via
- * - runBlocking(errorCollectorContextElement) { ... }
- * - runBlockingTest(Dispatchers.IO + errorCollectorContextElement) { ... }
+ * - `runBlocking(errorCollectorContextElement) { ... }`
+ * - `runTest(Dispatchers.IO + errorCollectorContextElement) { ... }`
  */
-val errorCollectorContextElement get() = ThreadLocalErrorCollector.instance.asContextElement()
+val errorCollectorContextElement: CoroutineContext.Element
+   get() = ErrorCollectorContextElement(threadLocalErrorCollector.get())
 
-class ThreadLocalErrorCollector : BasicErrorCollector() {
-   companion object {
-      val instance = object : ThreadLocal<ErrorCollector>() {
-         override fun initialValue() = ThreadLocalErrorCollector()
-      }
+
+private val threadLocalErrorCollector = object : ThreadLocal<CoroutineLocalErrorCollector>() {
+   override fun initialValue() = CoroutineLocalErrorCollector()
+}
+
+
+private class CoroutineLocalErrorCollector : BasicErrorCollector() {
```

Comment:
Do you think we should copy the parent, or just start a new one ?

Reply:
Copying the parent collector seems ideal. That way, we keep the parent clues and settings as context, like we do in sequential code, as shown in the test case:
```kotlin
clueContextAsString() shouldBe "$parentClue\n$childClue\n"
```

Seems like good companionship with structured concurrency.

Reply:
Great